doc/sphinx/develop/dotnet-extensions.md

```diff
@@ -0,0 +1,79 @@
+# Auto-Generated .NET API
+
+```{caution}
+The auto-generated .NET API is an experimental part of Cantera and may be changed or
+removed without notice.
+
+The current .NET API implementation is an early automated code generation prototype.
+It parses CLib header files rather than generating CLib specifications from Doxygen as
+described for the generic sourcegen [automated code generation](sec-sourcegen-details)
+approach. The language-specific source generation is incomplete but stable.
+```
+
+The .NET API is written in C# and supports .NET Standard 2.0 (for the primary project)
+and .NET 8 (and newer) on all platforms that support both .NET and the Cantera C++
+library. The .NET interface requires an installation of the .NET 8.0 SDK.
+
+(sec-sourcegen-dotnet-install)=
+## Building the .NET Interface
+
+After [building the main Cantera library](sec-compiling), switch to the
+`interfaces/dotnet` directory and run
+
+```bash
+dotnet build
+```
+
+The .NET test suite is invoked by running
+
+```bash
+dotnet test
+```
+
+In order to force re-import of generated code from sourcegen, a manual deletion of
+`obj` and `bin` folders in `Cantera`, `Cantera.Tests`, `examples/Applications` and
+`examples/Soundspeed` may be necessary.
+
+### C# Code Generation
+
+Source generation for the .NET interface is fully integrated into the build process.
+C# files used by the .NET API can be generated for informational purposes by running the
+following command from the root folder of the Cantera source code:
+
+```bash
+python interfaces/sourcegen/run.py --api=csharp --output=build/csharp
+```
+
+Auto-generated C# files are placed in the output folder `build/csharp`.
+
+## .NET Source Generator Overview
+
+The primary C# project is `Cantera.csproj`. This project uses P/Invoke extensively with
+the native Cantera library via the Cantera C interface (CLib), and wraps the low-level
+interfaces with classes and concepts familiar to a .NET developer. As part of the build
+process, it invokes [sourcegen](sourcegen) to scaffold the interop code and some of the
+code for the wrapper objects, such as simple properties which can mapped directly to
+CLib getter and setter functions. `Cantera.csproj` targets .NET Standard 2.0 and .NET 8.
+This project will be released as a NuGet package.
+
+`Cantera.Tests.csproj` contains the unit tests for the Cantera .NET library and targets
+.Net 8.
+
+The examples directory contains separate projects for each Cantera example. These will
+be packaged as a template for use with the .NET CLI, allowing developers to create a
+solution with all the examples using `dotnet new`.
+
+## Extending the .NET API
+
+Like all of Cantera, we welcome contributions to the .NET interface. Contributors should
+review general and C#-specific [coding standards](CONTRIBUTING). Please note that
```

Comment:
This could link to the relevant section of `develop/style-guidelines`, perhaps in addition to the generic contributing guidelines.